lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -824,7 +824,13 @@ sub investigate {
     my $investigation = $job->investigate;
     $self->respond_to(
         json => {json => $investigation},
-        html => {text => join("\n", map { "$_: $investigation->{$_}" } keys %$investigation)});
+        html => {
+            inline => join(
+                "\n",
+                map {
+                    "<tr><td style=\"vertical-align: top;\"><b>$_:</b></td><td><pre>$investigation->{$_}</pre></td><tr>"
+                } sort keys %$investigation
+            )});
```

Comment:
Is this really supposed to be an `inline` template? Do the variables contain Perl code and `<%...%>` blocks?

Reply:
no, they don't. `$investigation` is a hash with section headers as key and multi-line plain-text strings as value. All the HTML you can see in the line starting with "<tr>"

Reply:
Ok, then you don't want to use `inline` here, that's specifically for inline templates. You can serve `html` as `text` or `data` (depending on if you need the string to be `UTF-8` encoded).

Reply:
done